src/Diagnostics.RuntimeHost/Utilities/DateTimeHelper.cs

```diff
@@ -143,19 +143,20 @@ internal static bool PrepareStartEndTimeUtc(string startTime, string endTime, ou
 
             if (endTimeUtc > currentUtcTime.AddMinutes(kustoDelayInMinutes))
             {
-                /* Creteria for adjusting the time here:
+                /* Criteria for adjusting the time here:
                  *  1. forceAdjustStartAndEndTimes Flag set to True, or
-                 *  2. If Endtime is more than "currentTime - kustoDelay" and is very close the current time.
+                 *  2. If Endtime is more than "currentTime - kustoDelay" and is very close the current time, but less than or equal to the currentTime.
                  *          - We got a sev2 incident for this issue where we returned a bunch of BAD Requests to the clients.
                  *          - This can happen if client machine time is little bit off from the server machine time.
                  *          - We will only adjust if that offset is no more than 15 mins. If Client machine time is way off than server time, we will throw BAD REQUEST
                 */
-                if (forceAdjustStartAndEndTimes || (currentUtcTime - endTimeUtc).TotalMinutes < 15)
+                if (forceAdjustStartAndEndTimes || (endTimeUtc <= currentUtcTime && (currentUtcTime - endTimeUtc).TotalMinutes < 15))
                 {
-                    if(endTimeUtc - startTimeUtc < TimeSpan.FromHours(24))
+                    var duration = endTimeUtc - startTimeUtc;
+                    if (duration < TimeSpan.FromHours(24))
                     {
                         endTimeUtc = currentUtcTime.AddMinutes(kustoDelayInMinutes);
-                        startTimeUtc = endTimeUtc.AddHours((endTimeUtc - startTimeUtc).TotalHours * -1);
+                        startTimeUtc = endTimeUtc - duration;
```

Comment:
shouldnt this be duration.TotalHours ?

Reply:
The result is the same. The duration is a TimeSpan object, so you can directly add or subtract it without TotalHours or AddHours.